File: FYP.Portal/src/app/models/artist-with-features.ts
import { Artist } from './artist';
import { Genre } from './full-artist-stats';

export class UserRecommendedArtist {
    artist: Artist;
    features: SinglePopularityFeature[];
    matchedGenres: string[];
    genre: Genre;

    genreList = [];
    selected = false;
}

export class SinglePopularityFeature {
    score: number;
    term: string;
    averageMagnitude: number;
    rank: number;

    firstHalf = '';
    secondHalf = '';
}

File: FYP.Models/ArtistWithFeatures.cs
using System.Collections.Generic;

namespace FYP.Models
{
    public class ArtistWithFeatures
    {
        public Artist Artist { get; set; }
        public IList<SinglePopularityFeature> Features { get; set; }
    }

    public class UserRecommendedArtist
    {
        public Artist Artist { get; set; }
        public IList<SinglePopularityFeature> Features { get; set; }
        public IList<string> MatchedGenres { get; set; }
        public Genre Genre { get; set; }
    }
}